iomanagers/redis_manager.py

```diff
@@ -6,18 +6,28 @@
 
 
 class RedisManager:
-    def __init__(self, redis_host: str = 'localhost', redis_port: str = '6379', db_idx: int = 0):
+    def __init__(self, redis_host: str = 'localhost', redis_port: int = 6379, db_idx: int = 0):
         # decode_responses equals True makes sure the response is composed of strings instead of bytes
-        self.redis_client = redis.Redis(redis_host, redis_port, db_idx, decode_responses=True)
+        self.client = redis.Redis(redis_host, redis_port, db_idx, decode_responses=True)
+        self.key_expiration_time = 3600
+        self.search_history_max_length = 10
 
-    # save search results as Redis list
     def cache_search_results(self, search_term: str, search_results: list) -> None:
         if search_results is None or len(search_results) == 0:
             return
         redis_key = "search_term:" + search_term
-        self.redis_client.rpush(redis_key, *search_results)
+        self.client.rpush(redis_key, *search_results)
+        self.client.expire(redis_key, self.key_expiration_time)
 
-    # retrieve search results from Redis
     def get_search_results(self, search_term: str) -> list:
         redis_key = "search_term:" + search_term
-        return self.redis_client.lrange(redis_key, 0, -1)
+        return self.client.lrange(redis_key, 0, -1)
+
```

Comment:
[nit] may be start having function docstrings and args have typehints https://docs.python.org/3/library/typing.html

Reply:
Will do in another PR. We need to add mocking.